backend/src/main/java/pl/zzpj/esportbetting/impl/UserDetailsServiceImpl.java

```diff
@@ -60,6 +62,12 @@ public User findById(long id) {
                 () -> new ObjectNotFoundException("Not found user with id: " + id));
     }
 
+    @Scheduled(cron = "0 0 10 ? * Mon")
+    public void giveWeeklyCoins(){
+        for(User user: userRepository.findAll()){
+            user.setCoins(user.getCoins() + WEEKLY_COINS);
```

Comment:
jedyna uwaga: proponowałbym zaminić setCoins na addCoins, bo z setterem tak jest że można czasem zapomnieć dać getCoins i robi sie bigos xD

Reply:
tak to bez uwag :>

Reply:
poprawiłem xd